src/main/java/entity/data_structure/DataStore.java

```diff
@@ -0,0 +1,47 @@
+package entity.data_structure;
+
+/**
+ * An abstract data structure.
+ * @param <T> an object for the data structure.
+ */
+public interface DataStore<T> {
+    /**
+     * Adds element to data structure.
+     * @param element value to add to list
+     */
+    void add(T element);
+
+    /**
+     * Removes element from data structure.
+     * @param element element to remove.
+     */
+    void remove(T element);
+
+    /**
+     * Gets element from data structure.
+     * @param element element to get.
+     * @return returns the element.
+     */
+    T get(T element);
+
+    /**
+     * Removes and returns the element that has been removed.
+     * @param element element to pop.
+     * @return element that has been popped.
+     */
+    T pop(T element);
```

Comment:
Confusing Javadoc: Needs more detail. 

What is the difference between pop and remove? Isn't pop typically used to remove the last method of the data structure and return it (kind of like in a queue)?

Reply:
Also if both do the same thing (except one just returns the removed item), then why not just keep the pop method and get rid of the remove method?

Then, if I am using the pop method and don't need the returned item, I simply don't store it in a variable.